imagetool/src/main/java/com/oracle/weblogic/imagetool/api/model/AbstractFile.java

```diff
@@ -29,7 +29,13 @@ public AbstractFile(String id, String version, CachePolicy cachePolicy, String u
     }
 
     public static String generateKey(String id, String version) {
-        return id + CacheStore.CACHE_KEY_SEPARATOR + version;
+        String key = id;
+        if (id.indexOf('_') < 0) {
+            if (version != null) {
+                key = key + CacheStore.CACHE_KEY_SEPARATOR + version;
+            }
+        }
+        return key;
```

Comment:
Are you intentionally allowing a key without a version suffix?  If "id" does not have an underscore, then there is no version in the key.  For what case is that valid?

Reply:
Actually I already removed the version number (the code should just return the id),  the doc asked the user to specify the key in the format of id_releaseNumber.

Also, the addPatch is for manually adding the patch to the cache store and the key will be the id.

If the user then use off line mode for the imagetool,  then it still works since the look up is by the key only.

If it is online mode, then it won't find it in the cache and it will download again and set the right key with the release number.  The reason I want to do this is everywhere use specify patchId will now be in the format of patchid_releasenumber and not patchid.